src/api_server/src/lib.rs

```diff
@@ -217,6 +231,19 @@ impl ApiServer {
         response
     }
 
+    fn check_for_fatal_error(&self, response: &std::result::Result<VmmData, VmmActionError>) {
+        if let Err(e) = response {
+            let mut is_fatal = self.encountered_fatal_error.borrow_mut();
+            // Errors considered as fatal are added to this 'match'
+            match e {
```

Comment:
save this `e` to log it before exiting.

Reply:
What do you think if I create a separate structure containing both the error and an exit code meaningful to it? This way I could log both the error and pass the exit code to libc::exit.